### PR TITLE
Add support for the 'note' command to save new notes and edits

### PR DESCRIPTION
**Pull request by shawnnygoh**

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -85,4 +88,40 @@ public GuiSettings getGuiSettings() {
...
     }
+
+    @Override
+    public Storage getStorage() {
+        return storage;
+    }
+
+    @Override
+    public String readNote(Person person) throws IOException {
+        return storage.readNote(person);
```
Comment (tim0tay) [inline src/main/java/seedu/address/logic/LogicManager.java:99]: Shawn, I love the SLAPping that's going on here. It really makes the code much cleaner by utilising the abstraction principle!

File: src/main/java/seedu/address/storage/FileNotesStorage.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import java.nio.file.Path;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.person.Person;
+
+/**
+ * A class to manage notes storage in the file system.
+ */
+public class FileNotesStorage implements NotesStorage {
```
Comment (tim0tay) [inline src/main/java/seedu/address/storage/FileNotesStorage.java:14]: I really appreciate the use of an interface and implementing it via FileNotesStorage in alignment with the Dependency Inversion Principle!

**Review by tim0tay — APPROVED** [review]
Good job with all of these Shawn, especially with the brilliant thinking on how to implement this window and saving changes, whilst handling edge cases. A true champ.🥇  LGTM

File: src/test/java/seedu/address/storage/FileNotesStorageTest.java
Comment (wentingchua) [inline src/test/java/seedu/address/storage/FileNotesStorageTest.java:1]: Such comprehensive test cases! Good job Shawn! 👍

File: src/main/java/seedu/address/ui/NoteWindow.java [hunk 3]
```diff
@@ -34,6 +41,57 @@ public NoteWindow() {
...
+     * Creates a new NoteWindow for a specific person.
+     *
+     * @param person The person to create notes for
+     */
+    public NoteWindow(Person person, Logic logic) {
+        this();
+        this.person = person;
+        this.logic = logic;
+        if (person != null) {
+            getRoot().setTitle("Notes for " + person.getName().fullName);
```
Comment (wentingchua) [inline src/main/java/seedu/address/ui/NoteWindow.java:54]: Wow, conscientious effort in UI/UX design by making sure that the user knows which note window they are on! Great job, Shawn! 👍

**Review by wentingchua — APPROVED** [review]
LGTM! Thanks Shawn!
